components/supervisor/pkg/ports/ports-config.go

```diff
@@ -184,7 +186,7 @@ func parseWorkspaceConfigs(ports []*gitpod.PortConfig) (portConfigs map[uint32]*
 }
 
 func parseInstanceConfigs(ports []*gitpod.PortsItems) (portConfigs map[uint32]*gitpod.PortConfig, rangeConfigs []*RangeConfig) {
```

Comment:
Is this a concern around the number of bytes needed to store or some other size? I'd expect that the resulting size in bytes would be nearly identical, we're storing n items in both cases.

The extra property of Index probably costs us more as its repeated on each item as opposed to have the sorting implicit which is what a list would do.

Anyhow, just to wanted to flag this up as it felt odd.

Reply:
```yaml
# user can define ranged port like this, and ignore if this num of port is validate or not
ports:
  - port: 100001
  - port: 1-100000
```

With `Sort` property defined, we don't need anything. 

But if we use a new variable like `SortedPorts` array, it will contain `100001` items. Or, you need to check if this port number is inside this range or not, it makes code harder to read and wastes resources while getting the ports list

Reply:
We could store it as 
```
ports: [{range: {from: 1, to: 10000}, ...}, {range: {from: 10001, to: 10001}, ...}
```
Which would give us the same result.

Reply:
> Or, you need to check if this port number is inside this range or not everytime

It's this case

Reply:
Let's leave it open, and make it a follow-up PR if we are going to change it.

cc @akosyakov @iQQBot